### PR TITLE
Enable custom failure logging

### PR DESCRIPTION
**Pull request by satishpasumarthi**
*Issue #, if available:*
https://github.com/aws/sagemaker-training-toolkit/issues/111

*Description of changes:*
The current state of SM Training toolkit doesn't allow users to write their own custom failure messages in the failure file. This PR fixes this issue by allowing users to do the above. Also, we have made capture error to True by default.
*Testing done:*
Unit tests and Integration tests passed locally.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-training-toolkit/blob/master/README.md)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by satishpasumarthi** [comment]
@josephevans Could you please review this PR?

File: src/sagemaker_training/files.py [hunk 1]
```diff
@@ -48,7 +50,11 @@ def write_failure_file(failure_msg):  # type: (str) -> None
         failure_msg: The description of failure.
     """
     file_path = os.path.join(environment.output_dir, "failure")
-    write_file(file_path, failure_msg)
+
+    # Write failure file only if it does not exist
+    if not os.path.exists(file_path):
+        logger.info("Failure file already exists. Skipping creation....")
```
Comment (josephevans) [inline src/sagemaker_training/files.py:56]: Shouldn't this be logged when the above returns true?

**Review by josephevans — CHANGES_REQUESTED** [review]

File: src/sagemaker_training/files.py (hunk 1 above)
Comment (satishpasumarthi) [reply src/sagemaker_training/files.py]: Updated.

**Review by josephevans — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**